File: app/src/main/java/com/mytlogos/enterprise/viewmodel/FilterableViewModel.java
package com.mytlogos.enterprise.viewmodel;

import android.app.Application;

public abstract class FilterableViewModel extends RepoViewModel {

    FilterableViewModel(Application application) {
        super(application);
    }

    String processStringFilter(String filter) {
        if (filter != null && filter.isEmpty()) {
            filter = null;
        }
        if (filter != null) {
            filter = filter.toLowerCase();
        }
        return filter;
    }

    public abstract void resetFilter();
}

File: app/src/main/java/com/mytlogos/enterprise/viewmodel/FilterableViewModel.kt
package com.mytlogos.enterprise.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel

abstract class FilterableViewModel internal constructor(application: Application) :
    AndroidViewModel(application) {
    fun processStringFilter(filter: String): String {
        return filter.lowercase()
    }

    abstract fun resetFilter()
}